README.md

```diff
@@ -5,3 +5,312 @@
+  * [Deleting a feature](#Deleting-a-feature)
+* [Subscribing and handling messages](#Subscribing-and-handling-messages)
+* [Logging](#Logging)
+
+## Installation
+
+Install the sources and execute the following command:
+
+```commandline
+make install
+```
+
+## Creating and connecting a client
+
+Each client instance can optionally have a defined behaviour after connecting to or disconnecting from the client.
+
+```python    
+def on_connect(ditto_client: Client):
+    print("Ditto client connected")
+
+def on_disconnect(ditto_client: Client):
+    print("Ditto client disconnected")
+```
+
+With these configurations a client instance can be created.
+
+```python
+client = Client(on_connect=on_connect, on_disconnect=on_disconnect)
+```
+
+After the client is created, it's ready to be connected.
+
+```python
+client.connect("localhost")
+while True:
+    try:
+        time.sleep(5)
+    except KeyboardInterrupt:
+        print("finished")
+        client.disconnect()
+        sys.exit()
+```
+
+Full example of the basic client connection can be found [here](examples/client_connect.py).
+
+### Creating a client instance as a class
+
+The client can be created by inheriting the Client class. The `on_connect` and `on_disconnect` callback methods are overridden in order to be configured. A separate method is created in order to connect the client.
+
+```python
+class MyClient(Client):
+    def on_connect(self, ditto_client: Client):
+        print("Ditto client connected")
+
+    def on_disconnect(self, ditto_client: Client):
+        print("Ditto client disconnected")
+
+    def run(self):
+        self.connect("localhost", 1883)
+        while True:
+            try:
+                time.sleep(5)
+            except KeyboardInterrupt:
+                print("finished")
+                self.disconnect()
+                sys.exit()
+```
+
+After the client class is created, an instance of `MyClient` can be created and connected
+
+```python
+ditto_client = MyClient()
+ditto_client.run()
+```
+
+Full example of the client connection as a class can be found [here](examples/client_connect_as_class.py).
+
+### Connecting an external paho client
+
+It is also possible to create a client instance using external paho client, which allows adding custom topics and messages that are not supported in Ditto.
+
+A custom client class is created by inheriting the Client class.
+
+```python
+class MyClient(Client):
+    def on_connect(self, ditto_client: Client):
+        print("Ditto client connected")
+        self.subscribe(self.on_message)
+
+    def on_disconnect(self, ditto_client: Client):
+        print("Ditto client disconnected")
+        self.unsubscribe(self.on_message)
+
+    def on_message(self, request_id: str, message: Envelope):
+        print("request_id: {}, envelope: {}".format(request_id, message.to_ditto_dict()))
+
+    def on_log(self, ditto_client: Client, level, string):
+        print("[{}] {}".format(level, string))
```

Comment:
Could be narrowed down and included in the connection configuration options in the beginning.